Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -218,11 +241,20 @@ extension AlertStore {
     }
     struct SinceDateFilter: QueryFilter {
         let date: Date
-        var predicate: NSPredicate? { NSPredicate(format: "issuedDate >= %@", date as NSDate) }
+        let now: Date
+        let excludingFutureAlerts: Bool
+        var predicate: NSPredicate? {
+            let datePredicate = NSPredicate(format: "issuedDate >= %@", date as NSDate)
+            // if case .delayed(let interval) = $0.trigger, $0.issuedDate + interval >= date {
+            let futurePredicate = NSPredicate(format: "triggerInterval == nil OR (triggerType == 1 AND CAST(issuedDate, 'NSNumber') + triggerInterval < CAST(%@, 'NSNumber'))", now as NSDate)
+            return excludingFutureAlerts ?
+                NSCompoundPredicate(andPredicateWithSubpredicates: [datePredicate, futurePredicate])
+                : datePredicate
+        }
     }
 
-    func executeQuery(since date: Date, limit: Int, completion: @escaping (QueryResult<SinceDateFilter>) -> Void) {
-        executeAlertQuery(from: QueryAnchor(filter: SinceDateFilter(date: date)), limit: limit, completion: completion)
+    func executeQuery(since date: Date, excludingFutureAlerts: Bool = true, now: Date = Date(), limit: Int, completion: @escaping (QueryResult<SinceDateFilter>) -> Void) {
```

Comment:
[nit] Any reason you are swapping the order of these parameters (`now` and `excludingFutureAlerts`)?